### PR TITLE
PF-2 Static Checking and Code Review

### PR DESCRIPTION
**Pull request by meta-ramjilal-choudhary**

### CODE REVIEW

File: PF/PF-Session2/HexaDecimalCode/HexCalc.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	HexCalc(){
+		this.BASE = 16;
+	}
+	
+	/*
+	 * @param   two hexadecimal string number
+	 * @return  Addition of two hexadecimal number in string form
+	 */
+	public String add(String number1, String number2) {
+		long integerNumber1 = Long.parseLong(this.anyBaseToDecimal(number1));
```
Comment (metaamitasharma) [inline PF/PF-Session2/HexaDecimalCode/HexCalc.java]: Verify numbers first against hex. Add a method to validate input Hexadecimal.

File: PF/PF-Session2/HexaDecimalCode/HexCalc.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	/*
+	 * @param   Two hexadecimal number string
+	 * @return  Division of Two hexadecimal number
+	 */
+	public String divide(String number1, String number2) throws ArithmeticException {
+		long integerNumber1 = Long.parseLong(this.anyBaseToDecimal(number1));
+		long integerNumber2 = Long.parseLong(this.anyBaseToDecimal(number2));
+		
+		if(integerNumber2 == 0) {
+			throw new ArithmeticException();
```
Comment (metaamitasharma) [inline PF/PF-Session2/HexaDecimalCode/HexCalc.java]: Add error message to exception.

File: PF/PF-Session2/HexaDecimalCode/HexCalc.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	 * @return  -  0  if both are equal
+	 * 			-  1  if number1 > number2
+	 * 			- -1  if number1 < number2
+	 */
+	public int compare(String number1, String number2) {
+		
+		long integerNumber1 = Long.parseLong(this.anyBaseToDecimal(number1));
+		long integerNumber2 = Long.parseLong(this.anyBaseToDecimal(number2));
+		
+		if (integerNumber1 > integerNumber2) {
```
Comment (metaamitasharma) [inline PF/PF-Session2/HexaDecimalCode/HexCalc.java]: integerNumber1 > integerNumber2 ? return 1 : (integerNumber1 < integerNumber2 ? return -1 : return 0);

File: PF/PF-Session2/FCFS_Code/Process.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	}
+	
+	/*
+	 * Computer Waiting Time for a Process
+	 * @param   nothing
+	 * @return  nothing
+	 */
+	public void setWaitingTime() {
+		
+		this.waitingTime = (this.turnAroundTime > this.burstTime) ? (this.turnAroundTime - this.burstTime) : (this.burstTime - this.turnAroundTime) ;
```
Comment (metaamitasharma) [inline PF/PF-Session2/FCFS_Code/Process.java]: Is there any possibility that turnAroundTime can be less than burstTime ?

File: PF/PF-Session2/FCFS_Code/Process.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	}
+	
+	/*
+	 * Compute Turn Around Time
+	 * @param   nothing
+	 * @return  nothing
+	 */
+	public void setTurnAroundTime() {
+		
+		this.turnAroundTime = (this.completionTime > this.arrivalTime) ? (this.completionTime - this.arrivalTime) : (this.arrivalTime - this.completionTime) ;
```
Comment (metaamitasharma) [inline PF/PF-Session2/FCFS_Code/Process.java]: Is there possibility that completionTime is less than arrivalTime ?

File: PF/PF-Session2/FCFS_Code/FCFSScheduling.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+		this.maximumWaitingTime = 0;
+	}
+	
+	/*
+	 * Computes Completion time, Turn Around time, Waiting time
+	 * @param   numberOfProcess - Total number of Process
+	 * @param   arrivalBurstTime - Two dimensional array with arrival time and burst time
+	 * @return  nothing
+	 */
+	public void computationOFTimes(Process process[], int numberOfProcess, long arrivalBurstTime[][]) {
```
Comment (metaamitasharma) [inline PF/PF-Session2/FCFS_Code/FCFSScheduling.java]: Can we reduce arguments ? numberOfProcess can be calculated by array size.

Comment (metaamitasharma) [reply PF/PF-Session2/FCFS_Code/FCFSScheduling.java]: No need to pass process array.Reference  question.

File: PF/PF-Session2/FCFS_Code/FCFSScheduling.java [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+	public long getMaximumWaitingTime() {
+		return this.maximumWaitingTime;
+	}
+	
+	/*
+	 * Print list of Process 
+	 * @param process - Array of Process objects 
+	 * @param numberOfProcess - Total number of Process
+	 */
+	public void listProcess(Process process[], int numberOfProcess) {
```
Comment (metaamitasharma) [inline PF/PF-Session2/FCFS_Code/FCFSScheduling.java]: again, no need to pass numberOfProcess argument, can be taken from process[] array size.

File: PF/PF-Session2/HexaDecimalCode/HexCalc.java (hunk 1 above)
Comment (meta-ramjilal-choudhary) [inline PF/PF-Session2/HexaDecimalCode/HexCalc.java]: Create Method for validation and 
Verfiy in MainClass.java

File: PF/PF-Session2/HexaDecimalCode/HexCalc.java [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+			int ascii = (int) character ;
+			if( ascii >= 65 && ascii <= 70 ) {
+				integerNumber += (10 + ( ascii - 65 ))* Math.pow(base, power);
+			}
+			else {
+				integerNumber += (ascii - 48)*Math.pow(base, power);
+			}
+			power ++;
+		}
+		return Integer.toString(integerNumber);
```
[the remark this answers is not in the document]
Comment (meta-ramjilal-choudhary) [reply PF/PF-Session2/HexaDecimalCode/HexCalc.java]: Return type of method is String , So i convert Integer in String than return it.

File: PF/PF-Session2/HexaDecimalCode/HexCalc.java [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+	}
+	
+	/*
+	 * @param   Decimal number in String form
+	 * @return  Hexadecimal string 
+	 * Convert decimal to hexadecimal number
+	 */
+	public String decimalToAnyBase(String number) {
```
[the remark this answers is not in the document]
Comment (meta-ramjilal-choudhary) [reply PF/PF-Session2/HexaDecimalCode/HexCalc.java]: Input type is String , So i used string

**Comment by meta-ramjilal-choudhary** [comment]
Complete updated !

**Pull request merged**